app/models/state_change.rb

```diff
@@ -53,7 +53,7 @@ def record_current_state_of_target
   end
 
   def state_changer
-    target.state_changer
+    target&.state_changer
```

Comment:
A target shouldn't able to be nil based on the validations. Was there a specific case this was needed to be conditionally chained?

Reply:
This method is called by a validator and causes the model to throw an exception that causes a 500 error when using the API endpoint.  The target cannot be `nil` and the validations pass, but the validations all get run before we can determine that the model is broken.

Reply:
Can we do that?  It would probably make this code less confusing.

Reply:
The validation line currently just says:

```
validates :state_changer, presence: { message: 'target does not have a configured state changer' }
```

Reply:
Yep we can! https://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless